### PR TITLE
WRN-1346: Migrate Checkbox and CheckboxItem unit tests to Testing Library

### PR DESCRIPTION
**Pull request by adrian-cocoara-lgp**
Enact-DCO-1.0-Signed-off-by: Adrian Cocoara adrian.cocoara@lgepartner.com

### Checklist

* [x] I have read and understand the [contribution guide](http://enactjs.com/docs/developer-guide/contributing/)
* [ ] A [CHANGELOG entry](http://enactjs.com/docs/developer-guide/contributing/changelogs/) is included
* [ ] At least one test case is included for this feature or bug fix
* [x] Documentation was added or is not needed
* [ ] This is an API breaking change

### Issue Resolved / Feature Added
[//]: # (Describe the issue resolved or feature added by this pull request)
Migrated `Checkbox` and `CheckboxItem` unit tests from Enzyme to React Testing Library

### Links
[//]: # (Related issues, references)
WRN-1346

### CODE REVIEW

File: Checkbox/tests/Checkbox-specs.js [hunk 1]
```diff
@@ -1,61 +1,64 @@
...
+		const checkboxElement = getByRole('checkbox');
 
-		const expected = css.selected;
-		const actual = subject.prop('className');
+		expect(checkboxElement.className).not.toContain('selected');
+	});
 
-		expect(actual).not.toContain(expected);
+	test('should not be checked', () => {
+		const {getByRole} = render(<CheckboxBase />);
```
Comment (stanca-pop-lgp) [inline Checkbox/tests/Checkbox-specs.js:15]: Please leave an empty line between const and expect.

File: Checkbox/tests/Checkbox-specs.js [hunk 2]
```diff
@@ -1,61 +1,64 @@
...
+		const {getByRole} = render(<CheckboxBase selected />);
+		expect(getByRole('checkbox', {checked: true}));
 	});
 
-	test('should not include the indeterminate class when not indeterminate', () => {
-		const subject = shallow(
-			<CheckboxBase />
-		);
+	test('should add the indeterminate class when given the indeterminate prop', () => {
+		const {getByRole} = render(<CheckboxBase indeterminate />);
```
Comment (stanca-pop-lgp) [inline Checkbox/tests/Checkbox-specs.js:32]: Here also.

File: Checkbox/tests/Checkbox-specs.js [hunk 3]
```diff
@@ -1,61 +1,64 @@
...
+		expect(getByRole('checkbox').className).toContain('indeterminate');
+	});
 
-		const expected = css.indeterminate;
-		const actual = subject.prop('className');
+	test('should not include the indeterminate class when not indeterminate', () => {
+		const {getByRole} = render(<CheckboxBase />);
```
Comment (stanca-pop-lgp) [inline Checkbox/tests/Checkbox-specs.js:37]: Idem.

File: Checkbox/tests/Checkbox-specs.js [hunk 4]
```diff
@@ -1,61 +1,64 @@
...
+		expect(getByRole('checkbox').className).not.toContain('indeterminate');
+	});
 
-		expect(actual).not.toContain(expected);
+	test('should prioritize indeterminate over selected', () => {
+		const {getByRole} = render(<CheckboxBase indeterminate selected indeterminateIcon="Ind" />);
```
Comment (stanca-pop-lgp) [inline Checkbox/tests/Checkbox-specs.js:42]: Sort props.

File: CheckboxItem/tests/CheckboxItem-specs.js [hunk 5]
```diff
@@ -1,18 +1,45 @@
...
+	test('should have correct text', () => {
+		const {getByText} = render(<CheckboxItemBase>Hello CheckboxItem</CheckboxItemBase>);
+
+		const element = getByText(/Hello CheckboxItem/i);
+		expect(element).toBeInTheDocument();
+	});
+
+	test('should select with click', () => {
+		const {getAllByRole} = render(<CheckboxItem>Hello CheckboxItem</CheckboxItem>);
+
```
Comment (stanca-pop-lgp) [inline CheckboxItem/tests/CheckboxItem-specs.js:27]: Delete empty line.

**Review by alexandrumorariu — APPROVED** [review]
LGTM

**Review by stanca-pop-lgp — APPROVED** [review]
LGTM

**Pull request merged**